clients/src/main/java/org/apache/kafka/clients/consumer/internals/RequestFuture.java

```diff
@@ -37,44 +37,48 @@
  *
  * @param <T> Return type of the result (Can be Void if there is no response)
  */
-public class RequestFuture<T> {
-
-    private boolean isDone = false;
-    private T value;
-    private RuntimeException exception;
-    private List<RequestFutureListener<T>> listeners = new ArrayList<>();
+public class RequestFuture<T> implements ConsumerNetworkClient.PollCondition {
 
+    private static final Object NULL_SENTINEL = new Object();
```

Comment:
This is a nitpick, but wouldn't it be a bit clearer if we used a sentinel for the unset case and any value that is not the sentinel is considered OK?